### DISCUSSION TITLE
kbengine bbs data（Extracted by robot）

### DISCUSSION THREAD

**Discussion by datemore111** [kbengine/kbengine#1367] [General]
https://github.com/datemore111/kbengine_bbs_post.git

[kbengine_post.txt](https://github.com/kbengine/kbengine/files/5957799/kbengine_post.txt)

**Comment by datemore111** [comment]
# author：kbe, date: 2015-02-12 09:32:40, subject: kbengine新手必看
KBEngine服务端引擎开源项目地址(github)：

[url=https://github.com/kbengine/kbengine]https://github.com/kbengine/kbengine[/url]



引擎下载与编译:

[url=http://kbengine.org/cn/docs/build.html]http://kbengine.org/cn/docs/build.html[/url]



安装:

[url=http://kbengine.org/cn/docs/installation.html]http://kbengine.org/cn/docs/installation.html[/url]



启动与关闭 :

[url=http://kbengine.org/cn/docs/startup_shutdown.html]http://kbengine.org/cn/docs/startup_shutdown.html[/url]



技术概览:

[url=http://kbengine.org/cn/docs/documentations/kbengine_overview.html]http://kbengine.org/cn/docs/documentations/kbengine_overview.html[/url]



kbengine API手册下载

[url=http://kbengine.org/assets/other/kbengine_api.chm]http://kbengine.org/assets/other/kbengine_api.chm[/url]

(下载后如果是空白页，请鼠标右键kbengine_api.chm打开属性面板， 面板上点击“解除锁定”。)



FAQ:

[url=http://kbengine.org/cn/docs/faq.html]http://kbengine.org/cn/docs/faq.html[/url]

[url=http://bbs.kbengine.org/forum.php?mod=viewthread&tid=4&extra=page%3D1]http://bbs.kbengine.org/forum.php?mod=viewthread&tid=4&extra=page%3D1[/url]



kbengine视频教程（疯狂的青蛙制作）

[url=http://bbs.kbengine.org/forum.php?mod=viewthread&tid=1546&extra=page%3D1]http://bbs.kbengine.org/forum.ph ... 1546&extra=page%3D1[/url]





服务端demo代码解析：

[url=http://bbs.kbengine.org/forum.php?mod=viewthread&tid=166]http://bbs.kbengine.org/forum.php?mod=viewthread&tid=166[/url]



压力测试视频:

[url=http://bbs.kbengine.org/forum.php?mod=viewthread&tid=4055&extra=page%3D1]http://bbs.kbengine.org/forum.ph ... 4055&extra=page%3D1[/url]



QQ官方交流群：

[color=#000][font=Simsun][size=3]461368412、[/size][/font][/color][color=#000][font=Simsun][size=3]16535321[/size][/font][/color]

### reply: cnsoft, date: 2015-02-12 15:01:40, 
:lol:lol:lol:lol:lol

### reply: Colin, date: 2015-02-12 15:29:18, 
好帖必须顶:lol

### reply: 欧亨利, date: 2015-02-12 15:54:53, 
支持，新手必看哦。:lol:lol:lol

### reply: iKBE, date: 2015-02-12 17:31:14, 
支持:)希望KB越做越好

### reply: pistol, date: 2015-02-13 17:55:43, 
支持一下~

### reply: xfan_hsl, date: 2015-02-13 23:39:01, 
mark 开始学习了

### reply: mrshan, date: 2015-02-14 10:01:18, 
顶啊顶，祝贺kbengine越办越好

### reply: 小鬼, date: 2015-03-03 15:27:26, 
顶了 非常感谢分享~

### reply: jxdeng, date: 2015-03-05 16:43:41, 
:lol 顶

### reply: jxdeng, date: 2015-03-05 16:44:25, 
支持一下，学习了。

### reply: Doby, date: 2015-03-11 09:10:19, 
:D:D:D:D 顶顶顶

### reply: jinpei618, date: 2015-03-18 23:38:13, 
支持，开始学习

### reply: lxqiang87, date: 2015-03-27 16:11:13, 
刚开始学习，希望越来越好

### reply: jinpei618, date: 2015-04-10 17:38:08, 
顶，开始好好学习kbe

### reply: graymelody, date: 2015-04-15 19:13:22, 
新人前来支持下:lol

### reply: peter, date: 2015-05-17 15:25:36, 
新人学习:victory:

### reply: thoreaw2005, date: 2015-06-17 11:07:49, 
kbengine API手册下载不能下载了



The requested URL /kbengine.org/assets/other/kbengine_api.chm was not found on this server.

### reply: kbe, date: 2015-06-17 12:31:06, 
引用：thoreaw2005 发表于 2015-6-17 11:07

kbengine API手册下载不能下载了



The requested URL /kbengine.org/assets/other/kbengine_api.chm was no ...

---




被链接到[url]http://bbs.kbengine.org/kbengine.org/assets/other/kbengine_api.chm[/url]去了，现在可以了

### reply: Gavin, date: 2015-07-15 15:11:54, 
顶！热爱服务器端开发。

### reply: admin, date: 2015-08-14 17:08:05, 
一些值得看的

[url]http://bbs.kbengine.org/forum.php?mod=viewthread&tid=4&extra=page%3D1[/url]

[url]http://bbs.kbengine.org/forum.php?mod=viewthread&tid=1&extra=page%3D1[/url]

[url]http://bbs.kbengine.org/forum.php?mod=viewthread&tid=534&extra=page%3D1%26filter%3Ddigest%26digest%3D1[/url]

[url]http://bbs.kbengine.org/forum.php?mod=viewthread&tid=166&extra=page%3D1[/url]

[url]http://bbs.kbengine.org/forum.php?mod=viewthread&tid=247&extra=page%3D1[/url]

### reply: 小笙, date: 2016-07-13 20:42:36, 
引用：kbe 发表于 2016-7-13 20:32

单独开贴问吧。

那是有意为之，一个基本类型保持一个实例就好了。

aliasname只针对复杂类型有用，主要作 ...

---




嗯，看了一个星期了，框架的世界观好大呀:'(，现在终于看得有些头绪了

### reply: 小笙, date: 2016-07-13 19:58:54, 
引用：admin 发表于 2016-7-13 18:45

不是问题， 是方案。

KBE的智能指针也是操作refcountable对象的，而refcountable是单独存在的，既然存在了 ...

---




OK  懂了 学习了

### reply: 小笙, date: 2016-07-13 20:00:04, 
alias.xml里的别名总是在覆盖，这是有意为之吗？如果有意的话，那么之前的别名有什么意义

### reply: 小笙, date: 2016-07-13 20:11:40, 
引用：小笙 发表于 2016-7-13 20:00

alias.xml里的别名总是在覆盖，这是有意为之吗？如果有意的话，那么之前的别名有什么意义 ...

---




有看见保存在datatypes是按照别名保存的，但是那个datatype里的aliasname就不是正确的

### reply: kbe, date: 2016-07-13 20:32:51, 
引用：小笙 发表于 2016-7-13 20:11

有看见保存在datatypes是按照别名保存的，但是那个datatype里的aliasname就不是正确的 ...

---




单独开贴问吧。

那是有意为之，一个基本类型保持一个实例就好了。

aliasname只针对复杂类型有用，主要作用是调试。

### reply: admin, date: 2016-07-13 18:45:39, 
引用：小笙 发表于 2016-7-13 17:42

历史问题是吧...  框架也有会历史问题 好吧

---




不是问题， 是方案。

KBE的智能指针也是操作refcountable对象的，而refcountable是单独存在的，既然存在了就有用的理由，手动性能高，智能则省事，但不至于是个问题。

虽然使用智能指针优雅一些， 但不妨碍refcountable自己存在并手动操作，在能不使用智能指针的地方几乎都没用智能指针，后面加入智能指针是解决一些相互依赖或者容易出错的情况。

另一个原因是KBE是一个针对产品的框架目标在于稳定性和产品的产出， 不是用于学术研究的，目前的代码几乎都是通过项目验证过的，只要一块代码稳定了不是特别说不过去的情况下是不会去改动的。

这也是目前没有引擎开发文档的原因。





有点像python这样的开源项目，而不是Ogre或者ACE这样的项目。

### reply: kbe, date: 2016-07-13 19:38:18, 
python的c层对象都是手动加减引用，一种方案而已。之所以说历史方案是因为这块确实可以改得更优美，其实还有人强烈建议都改成c++11语法。

### reply: 小笙, date: 2016-07-13 16:44:46, 
有的指针为什么不用智能指针呢，而要手动incRef，例如这代码

	(*propertyDescr)[attrName] = propertyDescription;

	(*propertyDescr_uidmap)[propertyDescription->getUType()] = propertyDescription;

	propertyDescription->incRef();

### reply: 小笙, date: 2016-07-13 17:42:35, 
引用：admin 发表于 2016-7-13 17:30

历史方案。

---




历史问题是吧...  框架也有会历史问题 好吧

### reply: admin, date: 2016-07-13 17:30:10, 
引用：小笙 发表于 2016-7-13 16:44

有的指针为什么不用智能指针呢，而要手动incRef，例如这代码

	(*propertyDescr)[attrName] = propertyDescr ...

---




历史方案。

### reply: XiaoChong_Huang, date: 2017-02-22 14:52:30, 
学习中，顶起来

### reply: kbe-py, date: 2017-02-28 17:08:17, 
顶，刚入门小白太需要了。

### reply: ywt155, date: 2017-05-24 08:26:45, 
支持，新手必看哦。:lol:lol:lol

### reply: leonard, date: 2017-10-20 12:14:47, 
网址都无法打开？

### reply: kbe, date: 2017-10-20 14:56:30, 
引用：leonard 发表于 2017-10-20 12:14

网址都无法打开？

---




我能开

### reply: sersoong, date: 2017-12-11 17:33:47, 
顶，加油kbe

### reply: 萧山机场, date: 2017-12-29 19:59:50, 
顶，加油kbe

### reply: kbe, date: 2018-01-15 09:58:46, 
：）

**Comment by datemore111** [comment]
---------------------------------------------------------------------------------------------------
author：roulang, date:2020-12-30 14:57:46, subject:貌似有人非法链接然后报了错
[root@izbp1115ogry6vybatghjgz manor]# .   baseapp01: [ERROR]: For a long time(3600s) not to empty the garbages, there may be a leak of the entitys(size:23), please use the "KBEngine.entities.garbages.items()" command query!

   baseapp01: [ERROR]: EndPoint::setupSSL: load SSL_CTX_use_certificate_file(key/server_cert.pem): error:02001002:system library:fopen:No such file or directory! check kbengine[_defs].xml->channelCommon->sslCertificate

   baseapp01: [ERROR]: PacketReader::processMessages: not found msgID=790, msglen=515, from 47.101.72.8:57048/0/0/0.

   baseapp01: [ERROR]: PacketReader::processMessages: not found msgID=17735, msglen=238, from 47.101.72.8:57056/0/0/0.

   baseapp01: [ERROR]: EndPoint::setupSSL: load SSL_CTX_use_certificate_file(key/server_cert.pem): error:20074002:BIO routines:FILE_CTRL:system lib! check kbengine[_defs].xml->channelCommon->sslCertificate

   baseapp01: [ERROR]: PacketReader::processMessages: not found msgID=790, msglen=515, from 47.101.72.8:57086/0/0/0.

   baseapp01: [ERROR]: PacketReader::processMessages: not found msgID=17735, msglen=238, from 47.101.72.8:57134/0/0/0.

reply: 柯标, date:2021-01-02 11:39:35, 
我这里没问题

reply: DarkWere, date:2021-01-02 13:18:44, 
I have the same problem.







---------------------------------------------------------------------------------------------------
author：ycq, date:2021-01-03 11:55:23, subject:无法找到 v142 的生成工具
大哥们 kbe4.24 master版本  vs2017 首次编译出现了这种情况 重定目标不管用 还有什么解决方案吗？必须装2019？ 大佬支支招



 还有我这vs2017找不到 常规- 平台程序集 是怎么回事 少安装什么吗？







---------------------------------------------------------------------------------------------------
author：suweitao8, date:2021-01-03 14:23:54, subject:皇室战争复刻
现在只完成了一些基本的功能，但大体上能玩[url]https://github.com/suweitao8/RoyaleWar[/url]

reply: 柯标, date:2021-01-03 18:30:02, 
装2019吧

reply: 柯标, date:2021-01-03 18:30:49, 
可以贴点图发点视频啊

**Comment by kebiao** [comment]
：）
